include/alarm_table_defs.hpp

```diff
@@ -61,12 +61,18 @@ class AlarmTableDef
 
   unsigned int state() const;
 
+  const std::string& name() const    {return _alarm_definition._name;}
   unsigned int alarm_index() const   {return _alarm_definition._index;} 
   AlarmDef::Cause cause() const      {return _alarm_definition._cause;}
 
-  AlarmDef::Severity severity() const    {return _severity_details._severity;}
-  const std::string& description() const {return _severity_details._description;}
-  const std::string& details() const     {return _severity_details._details;}
+  AlarmDef::Severity severity() const             {return _severity_details._severity;}
+  const std::string& description() const          {return _severity_details._description;}
+  const std::string& details() const              {return _severity_details._details;}
+  const std::string& extended_cause() const       {return _severity_details._cause;}
```

Comment:
To distinguish it from the other cause returning function. Each alarm has a cause like "SOFTWARE_ERROR" and then each severity within that alarm has an extended_cause like "The system has detected that the Sprout process has failed. Monit will automatically attempt to restart it."

I've just checked, I thought I was copying the name from the JSON parser but that actually called it detailed_cause. I can change the names for consistency.